dev/components/stock-popup.js

```diff
@@ -0,0 +1,217 @@
+    let openPrice = stockData.dataset_data.data[0][1];
+    let low = stockData.dataset_data.data[0][3];
+    let high = stockData.dataset_data.data[0][2];
+    let volume = stockData.dataset_data.data[0][5];
+
+    // render stock name
+    this.$stockName.text(companyName);
+
+    let row = `
+      <tr>
+        <td class="key">Close</td>
+        <td class="val">${closePrice}</td>
+      </tr>
+      <tr>
+        <td class="key">Open</td>
+        <td class="val">${openPrice}</td>
+      </tr>
+      <tr>
+        <td class="key">Day's Range</td>
+        <td class="val">${low} - ${high}</td>
+      </tr>
+      <tr>
+        <td class="key">Volume</td>
+        <td class="val">${volume}</td>
+      </tr>
+    `;
+    this.$tbody.append(row);
+
+    // add stock id to watchlist button
+    this.$watchlistButton.attr('id', `${companyId}`);
+  }
+
+
+  // RENDER GRAPH
+  renderGraph(companyId) {
+    const stockData = store.get(`${companyId}`);
+
+    // get opening prices for company stock
+    let priceData = this.getSpecificCompanyData(stockData, 1);
+
+    // get dates for the opening prices
+    let dateLabels = this.getSpecificCompanyData(stockData, 0);
+
+    // create new graph for this company stock
+    this.graph = new Graph(this.$chartContainer, priceData, dateLabels);
+  }
+
+
+  // GET SPECIFIC DATA ARRAY OF COMPANY (STOCK OPEN PRICES, DATES, ETC.)
+  getSpecificCompanyData(data, num) {
+    return data.dataset_data.data.slice(0, 30).map((day) => {
+      return day[num];
+    }).reverse();
+  }
+
+
+  // ADD STOCK TO WATCHLIST
+  addToWatchlist() {
+    // retrieve watchlist array from storage
+    let watchlist = store.get('watchlist') || []; // ['AAPL', 'AMD'] or []
+
+
+    this.$popupContentContainer.on('click', 'button', function(event) {
+      event.preventDefault();
+      const $this = $(this);
+      let id = this.id;
+      const hasStock = watchlist.includes(id);
+
+      // if stock doesn't exist in watchlist, then add to watchlist
+      if (hasStock === false) {
+        // add stock into watchlist array
+        watchlist.push(id);
+        // store updated watchlist array
+        store.set('watchlist', watchlist);
+
+        // update watchlist button to REMOVE
+        $this.addClass('is-remove');
+        $this.text('Remove from watchlist');
+      }
+      // if stock exist, then remove it from watchlist
+      else {
+        // remove stock from watchlist array
+        let index = watchlist.indexOf(id);
+        if (index != -1) {
+          watchlist.splice(index, 1);
+        }
+
+        // store upated watchlist array
+        store.set('watchlist', watchlist);
+        console.log('Removed from watchlist', watchlist);
+
+        // update watchlist button to ADD
+        $this.removeClass('is-remove');
+        $this.text('Add to watchlist');
+      }
+    });
+  }
+
+
+  // UPDATE WATCHLIST BUTTON STATE (ADD OR REMOVE)
+  activateWatchlistButtonState(companyId) {
```

Comment:
I created a method called toggleButtonState but I'm not exactly sure if I did it correctly. If you can look this over, that'd be awesome. Thanks!